src/main/java/com/books/service/BookService.java

```diff
@@ -25,63 +27,74 @@ public BookService(BookRepository repository) {
 
     /**
      * Persists a book object.
+     *
      * @param book book to be saved.
      */
-    public void saveBook(Book book){
+    public void saveBook(Book book) {
         repository.save(book);
     }
 
     /**
-     * Retrieves all books.
-     * @return all books.
+     * Retrieves all books in the current page with requested page size.
+     *
+     * @param pageNum Page number of request - zero based.
+     * @param entriesPerPage Number of entries per page.
+     * @return All books in the current page with requested page size.
      */
-    public List<Book> getAllBooks(){
-        return (List<Book>) repository.findAll();
+    public List<Book> getAllBooks(int pageNum, int entriesPerPage) {
```

Comment:
getAllBooks is misleading name, because it does not return all books but only books for 1 page. Rename to getBooksByPage of something like that

Reply:
Done